pkg/controller/atlasdeployment/serverless_private_endpoint.go

```diff
@@ -0,0 +1,356 @@
+		if p.Status != SPEStatusAvailable {
+			return workflow.Terminate(workflow.ServerlessPrivateEndpointReady, fmt.Sprintf("serverless private endpoint %s is not ready", p.ID))
+		}
+	}
+	return workflow.OK()
+}
+
+func deleteSPE(ctx context.Context, client mongodbatlas.ServerlessPrivateEndpointsService, groupID, deploymentName string, peToDelete []string) []error {
+	var result []error
+	for _, id := range peToDelete {
+		_, err := client.Delete(ctx, groupID, deploymentName, id)
+		if err != nil {
+			result = append(result, fmt.Errorf("failed to delete serverless private endpoint: %w", err))
+		}
+	}
+	return result
+}
+
+func getStatusFromReadySPE(pe []mongodbatlas.ServerlessPrivateEndpointConnection) []status.ServerlessPrivateEndpoint {
+	var result []status.ServerlessPrivateEndpoint
+	for _, endpoint := range pe {
+		result = append(result, status.SPEFromAtlas(endpoint))
+	}
+	return result
+}
+
+func connectSPE(ctx context.Context, logger *zap.SugaredLogger, client mongodbatlas.ServerlessPrivateEndpointsService, groupID, deploymentName string, providerName provider.ProviderName, pe []mongodbatlas.ServerlessPrivateEndpointConnection) []status.ServerlessPrivateEndpoint {
+	var result []status.ServerlessPrivateEndpoint
+	for _, endpoint := range pe {
+		id := endpoint.ID
+		req := mongodbatlas.ServerlessPrivateEndpointConnection{
+			PrivateEndpointIPAddress: endpoint.PrivateEndpointIPAddress,
+			CloudProviderEndpointID:  endpoint.CloudProviderEndpointID,
+			ProviderName:             string(providerName),
+		}
+		logger.Debugf("Connecting serverless private endpoint %s", id)
+		resultPE, _, err := client.Update(ctx, groupID, deploymentName, id, &req)
+		if err != nil {
+			logger.Errorf("Failed to connect serverless private endpoint %s: %v", id, err)
+			result = append(result, status.FailedToConnectSPE(endpoint, fmt.Sprintf("failed to connect serverless private endpoint: %s", err)))
+		} else {
+			logger.Debugf("Serverless private endpoint %s is connected", id)
+			result = append(result, status.SPEFromAtlas(*resultPE))
+		}
+	}
+	return result
+}
+
+func createSPE(ctx context.Context, logger *zap.SugaredLogger, client mongodbatlas.ServerlessPrivateEndpointsService, groupID, deploymentName string, pe []mdbv1.ServerlessPrivateEndpoint) []status.ServerlessPrivateEndpoint {
+	var result []status.ServerlessPrivateEndpoint
+	for _, endpoint := range pe {
+		created, _, err := client.Create(ctx, groupID, deploymentName, prepareForCreation(endpoint))
+		if err != nil {
+			logger.Errorf("Failed to create serverless private endpoint: %v, err: %v", prepareForCreation(endpoint), err)
+			result = append(result, status.FailedToCreateSPE(endpoint.Name, fmt.Sprintf("failed to create serverless private endpoint: %s", err)))
+		} else {
+			result = append(result, status.SPEFromAtlas(*created))
+		}
+	}
+	return result
+}
+
+func prepareForCreation(pe mdbv1.ServerlessPrivateEndpoint) *mongodbatlas.ServerlessPrivateEndpointConnection {
```

Comment:
I guess it would be better to rename it to something more self-explaining like:
```suggestion
func newServerlessEndpoint(pe mdbv1.ServerlessPrivateEndpoint) *mongodbatlas.ServerlessPrivateEndpointConnection {
```

Reply:
Done